src/components/encoding-pane/filter-shelf.tsx

```diff
@@ -60,24 +61,25 @@ class FilterShelfBase extends React.Component<FilterShelfProps, {}> {
   private renderFilterShelf(filter: RangeFilter | OneOfFilter, index: number) {
     const {fieldDefs, schema} = this.props;
     const fieldIndex = schema.fieldNames().indexOf(filter.field);
-    const domain = schema.domain(fieldDefs[fieldIndex] as FieldQuery);
+    const fieldDef = fieldDefs[fieldIndex];
+    const domain = schema.domain(fieldDef as FieldQuery);
     return (
-      <div styleName='filter-shelf' key={filter.field}>
+      <div styleName='filter-shelf' key={index}>
```

Comment:
If we want to support multiple filters for one field, we should use index other than field to avoid conflict.